test/scripts/feature-flags.js

```diff
@@ -0,0 +1,258 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const ROOT_DIR = '../..';
+
+const { assert } = require('chai');
+const cp = require('child_process');
+const path = require('path');
+const Promise = require(`${ROOT_DIR}/promise`);
+
+cp.execAsync = Promise.promisify(cp.exec);
+
+const redis = require(`${ROOT_DIR}/redis`)({
+  enabled: true,
+  host: process.env.REDIS_HOST || '127.0.0.1',
+  port: process.env.REDIS_PORT || 6379,
+  prefix: 'featureFlags:',
+  maxConnections: 2,
+  maxPending: 1,
+  minConnections: 1,
+}, {
+  error () {},
+  info () {},
+  warn () {},
+});
+
+const KEYS = {
+  current: 'current',
+  previous: 'previous'
+};
+
+const cwd = path.resolve(__dirname, ROOT_DIR);
+
+describe('scripts/feature-flags:', () => {
+  let current, previous;
+
+  beforeEach(async () => {
+    current = await redis.get(KEYS.current);
+    previous = await redis.get(KEYS.previous);
+    await redis.del(KEYS.current);
+    await redis.del(KEYS.previous);
+  });
+
+  afterEach(async () => {
+    if (current) {
+      await redis.set(KEYS.current, current);
+    } else {
+      await redis.del(KEYS.current);
+    }
+    if (previous) {
+      await redis.set(KEYS.previous, previous);
+    } else {
+      await redis.del(KEYS.previous);
+    }
+  });
+
+  after(() => redis.close());
+
+  it('read does not fail', () => {
+    return cp.execAsync('node scripts/feature-flags read', { cwd });
+  });
+
+  [ 'write', 'merge' ].forEach(command => {
+    it(`${command} does not fail`, () => {
+      return cp.execAsync(`echo '{}' | node scripts/feature-flags ${command}`, { cwd });
+    });
+
+    it(`${command} fails if stdin is not valid JSON`, () => {
+      return cp.execAsync(`echo "wibble" | node scripts/feature-flags ${command}`, { cwd })
+        .then(() => assert(false, 'script should have failed'), () => {});
+    });
+
+    it(`${command} fails if stdin contains unexpected key`, () => {
+      return cp.execAsync(`echo '{"wibble":{}}' | node scripts/feature-flags ${command}`, { cwd })
+        .then(() => assert(false, 'script should have failed'), () => {});
+    });
+
+    it(`${command} does not fail with valid communicationPrefLanguages`, () => {
+      return cp.execAsync(`echo '{"communicationPrefLanguages":["en","fr"]}' | node scripts/feature-flags ${command}`, { cwd });
```

Comment:
Can you add a test for two part locales such as [pt-br](https://github.com/mozilla/fxa-content-server/blob/7af51def9f7fde6acccf9ecf5dd797341570c0dd/app/scripts/lib/experiments/grouping-rules/communication-prefs.js#L23) and the other [regexp based tests](https://github.com/mozilla/fxa-content-server/blob/7af51def9f7fde6acccf9ecf5dd797341570c0dd/app/scripts/lib/experiments/grouping-rules/communication-prefs.js#L16)?